mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/loading/impl/HibernateOrmByTypeEntityLoader.java

```diff
@@ -23,13 +25,21 @@
 	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
 
 	private final Map<String, HibernateOrmComposableEntityLoader<? extends T>> delegatesByEntityName;
+	private final TimingSource timingSource;
 
-	HibernateOrmByTypeEntityLoader(Map<String, HibernateOrmComposableEntityLoader<? extends T>> delegatesByEntityName) {
+	HibernateOrmByTypeEntityLoader(Map<String, HibernateOrmComposableEntityLoader<? extends T>> delegatesByEntityName,
+			TimingSource timingSource) {
 		this.delegatesByEntityName = delegatesByEntityName;
+		this.timingSource = timingSource;
 	}
 
 	@Override
 	public List<T> loadBlocking(List<EntityReference> references, Integer timeout) {
+		TimeoutManager timeoutManager = null;
```

Comment:
Here, you could just pass the `TimeoutManager` to this method instead of the `Integer timeout`. I expect this will make a lot of code simpler and safer.